Cubical/Homotopy/HSpace.agda

```diff
@@ -71,6 +72,21 @@ HSpace-Π∙-Equiv : {ℓ : Level} (A : Pointed ℓ) → HSpace A ≃
                   Π∙ A (λ x → A →∙ (typ A , x)) (idfun∙ A)
 HSpace-Π∙-Equiv A = isoToEquiv (HSpace-Π∙-Iso A)
 
+{- every strongly homogeneous structure on A gives rise
+   to a left-invertible H-space structure, by a slight modification -}
+HSpace-homogeneous : {ℓ : Level} (A : Pointed ℓ) → isHomogeneous A → HSpace A
+HSpace-homogeneous A h = Iso.inv (HSpace-Π∙-Iso A) s
+  where
+    k : (x : typ A) → A ≃∙ (typ A , x)
+    k x = pointed-sip⁻ A (typ A , x) (sym (h (pt A)) ∙ h x)
+
+    k₀ : k (pt A) ≡ idEquiv∙ A
+    k₀ = cong′ (pointed-sip⁻ A A) (lCancel (h (pt A))) ∙ pointed-sip⁻-refl A
```

Comment:
This is the slow line, taking about 20 seconds on my machine to elaborate.

Reply:
Agda is busy with ```pointed-sip⁻-refl A``` for more than 20 seconds on my machine.

Reply:
Ah, no, I wasn't careful enough: Only the first 10 seconds or so are spend on the sip-part.

Reply:
I have no idea yet, why it is so slow. I might get back at it in a couple of hours.

Reply:
The problem seems to be that ```pointed-sip⁻-refl A``` has a huge normalform. Agda spends 10s on deserializing it and some more time on concatenating it after that.

Reply:
Thanks for looking into this. What's the solution in this case – making it abstract?

Reply:
I am currently trying, making this particular term abstract only seems to reduce the time somehwat, but it is still >10s.

Reply:
Aha - making both, pointed-sip-refl and pointed-sip abstract solves it. I'll check if that breaks something...

Reply:
My network is too bad (train..) to do a PR on your PR, so I just paste here what I did (at the end of Cubical/Homotopy/Pointed.agda):
```agda
pointed-sip-idEquiv∙ : (A : Pointed ℓ) → pointed-sip A A (idEquiv∙ A) ≡ refl
fst (pointed-sip-idEquiv∙ A i j) = uaIdEquiv i j
snd (pointed-sip-idEquiv∙ A i j) = glue {φ = i ∨ ~ j ∨ j} (λ _ → pt A) (pt A)

{-
  The following terms have huge normalforms, so they are abstract to avoid
  type checking speed problems, for example in

    Cubical.Homotopy.HSpace
-}
abstract
  pointed-sip⁻ : (A B : Pointed ℓ) → (A ≡ B) → A ≃[ PointedEquivStr ] B
  pointed-sip⁻ A B = invEq (pointedSIP A B)

  pointed-sip⁻-refl : (A : Pointed ℓ) → pointed-sip⁻ A A refl ≡ idEquiv∙ A
  pointed-sip⁻-refl A = sym (invEq (equivAdjointEquiv (pointedSIP A A)) (pointed-sip-idEquiv∙ A))

pointedEquivAction : EquivAction {ℓ} PointedStructure
pointedEquivAction e = e

pointedTransportStr : TransportStr {ℓ} pointedEquivAction
pointedTransportStr e s = sym (transportRefl _)
```
My (slow) notebook is still checking the library with this change...

Reply:
Library typechecks.